### PR TITLE
[Issue 9790] Support disabled the tenants/namespaces force deletion in the broker.conf

### LINKED ISSUE #9790: Support disabled the tenants/namespaces force deletion in the broker.conf
**Is your feature request related to a problem? Please describe.**
Currently, we added the tenants and namespaces force deletion feature in the broker, which allows users to force delete the topics under the namespace or tenants and also delete the namespace or tenant itself. This is a very influential operation, so it's better to provide a way for the Pulsar maintainers to enable or disable the tenants/namespaces force deletion

**Describe the solution you'd like**
Add new configs:

```
forceDeleteTenantAllowed=false
forceDeleteNamespaceAllowed=false
```

It's better to keep the default value to `false`.

related to https://github.com/apache/pulsar/pull/9677 and https://github.com/apache/pulsar/pull/7993

### PR DESCRIPTION
**Pull request by murong00**
### Motivation

Fixes #9790

### Modifications

- Add new configs below and keep the default value to `false`:
```
forceDeleteTenantAllowed=false
forceDeleteNamespaceAllowed=false
```
- Add unit tests for this change.

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java [hunk 1]
```diff
@@ -290,6 +290,12 @@ protected void internalDeleteTenant(AsyncResponse asyncResponse, String tenant)
     }
 
     protected void internalDeleteTenantForcefully(AsyncResponse asyncResponse, String tenant) {
+        if (!pulsar().getConfiguration().isForceDeleteTenantAllowed()) {
+            asyncResponse.resume(
+                    new RestException(Status.FORBIDDEN, "Broker doesn't allow forced deletion of tenants"));
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java:295]: The @ApiReponse use 403 to indicate if the request has admin permission, I think 405 is more reasonable here?

Comment (murong00) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java]: Thanks for the tips, done.

**Review by codelipenghui — APPROVED** [review]

**Review by sijie — APPROVED** [review]

**Comment by Anonymitaet** [comment]
@murong00 thanks for your great work! Shall the changes be documented to the [broker -configuration](http://pulsar.incubator.apache.org/docs/en/next/reference-configuration/#broker) section?
If so, could you please help add the docs accordingly? Thanks

**Comment by murong00** [comment]
@Anonymitaet Added the docs accordingly, please help to take a look when you are available.

File: site2/docs/reference-configuration.md [hunk 2]
```diff
@@ -185,6 +185,8 @@ internalListenerName|Specify the internal listener name for the broker.<br><br>*
 |brokerDeleteInactiveTopicsFrequencySeconds|  How often to check for inactive topics  |60|
 | brokerDeleteInactiveTopicsMode | Set the mode to delete inactive topics. <li> `delete_when_no_subscriptions`: delete the topic which has no subscriptions or active producers. <li> `delete_when_subscriptions_caught_up`: delete the topic whose subscriptions have no backlogs and which has no active producers or consumers. | `delete_when_no_subscriptions` |
 | brokerDeleteInactiveTopicsMaxInactiveDurationSeconds | Set the maximum duration for inactive topics. If it is not specified, the `brokerDeleteInactiveTopicsFrequencySeconds` parameter is adopted. | N/A |
+|forceDeleteTenantAllowed| Allow forced deletion of tenants.|false|
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:188]:
```suggestion
|forceDeleteTenantAllowed| Allow you to delete a tenant forcefully. |false|
```

File: conf/broker.conf [hunk 3]
```diff
@@ -165,6 +165,12 @@ brokerDeleteInactivePartitionedTopicMetadataEnabled=false
 # Topics that are inactive for longer than this value will be deleted
 brokerDeleteInactiveTopicsMaxInactiveDurationSeconds=
 
+# Allow forced deletion of tenants.
```
Comment (Anonymitaet) [inline conf/broker.conf:168]:
```suggestion
# Allow you to delete a tenant forcefully.
```

File: conf/broker.conf [hunk 4]
```diff
@@ -165,6 +165,12 @@ brokerDeleteInactivePartitionedTopicMetadataEnabled=false
...
+forceDeleteTenantAllowed=false
+
+# Allow forced deletion of namespaces.
```
Comment (Anonymitaet) [inline conf/broker.conf:171]:
```suggestion
# Allow you to delete a namespace forcefully.
```

File: site2/docs/reference-configuration.md [hunk 5]
```diff
@@ -185,6 +185,8 @@ internalListenerName|Specify the internal listener name for the broker.<br><br>*
...
+|forceDeleteNamespaceAllowed| Allow forced deletion of namespaces.|false|
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:189]:
```suggestion
|forceDeleteNamespaceAllowed| Allow you to delete a namespace forcefully. |false|
```

File: site2/docs/reference-configuration.md (hunk 2 above)
Comment (murong00) [reply site2/docs/reference-configuration.md]: Done.

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**